### PR TITLE
ZCS-12412: WCAG | Classic UI | Medium Complexity | Aria issues Part - 3

### PR DESCRIPTION
**Pull request by miteshsavani810**
- Fixed missing aria-label on filter dialog.
- Fixed missing aria-label on Edit contact dialog.

See the changes of zm-ajax https://github.com/Zimbra/zm-ajax/pull/105

### CODE REVIEW

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js [hunk 1]
```diff
@@ -203,6 +205,14 @@ function(parent, segment, i) {
 			// TODO: guard against badly specified message
 			select.addOption(formats[i].toPattern(), i == 0, values[i]);
 		};
+
+		var setAriaLabel = function() {
+			select.setAriaLabel(select._selectedValue == ZmFilterRule.GROUP_ANY ? "If any of the following conditions are met" : "If all of the following conditions are met");
```
Comment (miteshsavani810) [inline WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:210]: Notes: Will translate newly added text once PR got approved.
`If any of the following conditions are met` , `If all of the following conditions are met`,

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js [hunk 2]
```diff
@@ -963,7 +985,7 @@ function(rowId, isCondition) {
 		button.setData(ZmFilterRuleDialog.IS_CONDITION, isCondition);
 		button.setData(ZmFilterRuleDialog.DO_ADD, (b == "Plus"));
 		button.addSelectionListener(this._plusMinusLstnr);
-		button.setAriaLabel(b);
+		button.setAriaLabel(b == "Plus" ? "Add filter rule": "Remove filter rule");
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:988]: `b === "Plus"` ?

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js [hunk 3]
```diff
@@ -560,6 +571,11 @@ function(conf, field, options, rowData, testType, rowId) {
 		if (isMainSelect) {
 			select.setData(ZmFilterRuleDialog.IS_CONDITION, isCondition);
 			select.addChangeListener(this._rowChangeLstnr);
+			if (field == "subject") {
+				select.setAriaLabel("Header");
+			} else if (field == "name") {
+				select.setAriaLabel("Action");
+			}
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:578]: let's use `===` for strict comparison ?

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js [hunk 4]
```diff
@@ -578,6 +595,7 @@ function(conf, field, options, rowData, testType, rowId) {
 			else if (testType == ZmFilterRule.TEST_CONVERSATIONS || testType == ZmFilterRule.TEST_LIST  ||  testType == ZmFilterRule.TEST_BULK || testType == ZmFilterRule.TEST_IMPORTANCE || testType == ZmFilterRule.TEST_FLAGGED) {
 				select.addChangeListener(this._importanceChangeLstnr);
 			}
+			select.setAriaLabel("value");
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:598]: "Value"
although it won't be visible let's stick to using titlecase

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js [hunk 5]
```diff
@@ -588,6 +606,10 @@ function(conf, field, options, rowData, testType, rowId) {
 			else if (testType == ZmFilterRule.TEST_CONVERSATIONS || testType == ZmFilterRule.TEST_LIST ||  testType == ZmFilterRule.TEST_BULK || testType == ZmFilterRule.TEST_FLAGGED) {
 				select.setVisibility(false);
 			}
+			select.setAriaLabel("value type");
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:609]: "Value Type"

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js:988]: Done.

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js]: Done.

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js (hunk 4 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js]: Updated

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js (hunk 5 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js]: Updated

File: WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js [hunk 6]
```diff
@@ -1561,6 +1561,9 @@ ZmEditContactViewInputSelect = function(params) {
     if (this._input && (params.inputWidth || params.inputHeight)) {
         Dwt.setSize(this._input.getInputElement(), params.inputWidth, params.inputHeight);
     }
+	if (this._input && params.ariaInputLabel) {
+		this._input.getInputElement().setAttribute('aria-label', params.ariaInputLabel);
+	}
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js:1566]: conditions can be combined

Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js]: Done.

**Review by HimanshuSojitraSynacor — APPROVED** [review]

File: WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js]: Added translations.

**Review by HimanshuSojitraSynacor — APPROVED** [review]

File: WebRoot/js/zimbraMail/package/Startup1_1.js [hunk 7]
```diff
@@ -120,6 +120,7 @@ AjxPackage.require("ajax.dwt.widgets.DwtMessageDialog");
 AjxPackage.require("ajax.dwt.widgets.DwtOptionDialog");
 AjxPackage.require("ajax.dwt.widgets.DwtChooser");
 AjxPackage.require("ajax.dwt.widgets.DwtTimeSelect");
+AjxPackage.require("ajax.dwt.widgets.DwtSelect");
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/package/Startup1_1.js:123]: we should remove this package from Startup2.js package

Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/package/Startup1_1.js:123]: Done

**Review by silentsakky — APPROVED** [review]
